Classes/Controller/Backend/Order/DocumentController.php

```diff
@@ -26,19 +26,10 @@
 
 class DocumentController extends ActionController
 {
-    protected PersistenceManager $persistenceManager;
-
-    protected ItemRepository $itemRepository;
-
-    public function injectPersistenceManager(PersistenceManager $persistenceManager): void
-    {
-        $this->persistenceManager = $persistenceManager;
-    }
-
-    public function injectItemRepository(ItemRepository $itemRepository): void
-    {
-        $this->itemRepository = $itemRepository;
-    }
+    public function __construct(
+        protected readonly PersistenceManager $persistenceManager,
+        protected readonly ItemRepository $itemRepository
+    ) {}
```

Comment:
I understood that you do not want to use DI via `construct()` but now all dependencies are injected that way!? (Same for all other controller)

Reply:
I think we have misunderstood each other. I have nothing against dependency injection via the constructor method, but in abstract classes it's not so nice. I think if someone extends the DocumentController, for example, then it's quite OK for me that the Constructor arguments have to be passed via a parent::__construct().
This is still possible. I had already considered declaring all the controllers as final.

Reply:
Okay!